components/script/dom/htmlformelement.rs

```diff
@@ -1332,6 +1390,29 @@ impl<'a> FormSubmitter<'a> {
                 ),
         }
     }
+
+    // https://html.spec.whatwg.org/multipage/#concept-submit-button
+    fn is_submit_button(&self) -> bool {
+        match *self {
+            FormSubmitter::InputElement(input_element) => {
```

Comment:
I think here we can add a link to https://html.spec.whatwg.org/multipage/input.html#image-button-state-(type=image)

Reply:
this line handles both type=image and type=submit, I added the links